### DISCUSSION TITLE
Opening .sh files fails to execute properly

### DISCUSSION THREAD

**Discussion by PianothShaveck** [ghostty-org/ghostty#3802] [Feature Requests, Ideas] [answered]
When attempting to execute .sh scripts by double-clicking them in Ghostyy, the behavior differs from the default macOS Terminal, leading to errors. Below are the specific issues observed:
	1.	Improper handling of special characters:
	•	Issue: Ghostyy does not escape special characters (e.g., spaces, parentheses) in folder or file names.
	•	Example: Executing a script named `my script (test).sh` located in a directory like `~/Projects/My Scripts/` results in an error, unless all spaces and parentheses from both the directories and the file name are removed.
	•	Expected behavior: The default macOS Terminal correctly handles such characters, allowing the script to run without issues.
	2.	Inconsistent cd Command Behavior:
	•	Issue: When double-clicking a .sh file, using `cd example` does not navigate to `~/example` by default in Ghostyy.
	•	Observed behavior: The cd command fails to default to the home directory when executed via double-click.
	•	Workaround: Dragging and dropping the .sh file into an empty Ghostyy window allows cd to function as expected.
	•	Expected behavior: Similar to the default Terminal, `cd example` in a .sh file should navigate to `~/example` without requiring drag-and-drop.

These inconsistencies make Ghostyy less convenient compared to the default Terminal application in some cases. Users would need to modify their .sh scripts or adapt their workflow to accommodate these discrepancies, which is not ideal.

	•	Ghostyy should correctly escape special characters in file and directory names when executing scripts via double-click.
	•	The cd command should behave consistently with the default Terminal, navigating relative to the home directory unless specified otherwise.

Ghostyy Version 1.0.0 Build 8469 Commit 44d4062
macOS Sonoma 14.7.1

Additional Notes:
These issues not only diverge from the default Terminal’s behavior but also introduce inconsistencies within Ghostyy itself, as detailed in the second point above.

(In case you're wondering, yes, I made ChatGPT rewrite the issue just to fix the grammar and the format, however I promise it's a human reporting this issue)

**Comment by belizariogr** [comment]
Other problem that I got here is that it runs using "/bin/bash --noprofile " like in the follow command line:

```
/Users/belizario/dev/bg/scripts/mariadb-users.sh: line 1: mariadb: command not found
/Users/belizario/dev/bg/scripts/mariadb-users.sh: line 2: mariadb: command not found

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

Ghostty failed to launch the requested command:

/usr/bin/login -flp belizario /bin/bash --noprofile --norc -c exec -l /Users/belizario/dev/bg/scripts/mariadb-users.sh

Runtime: 42 ms

Press any key to close the window.
``` 

That leads to fail to get environment variables... and, in that case, cant get path that comes from .zshrc

**Comment by mitchellh** [comment, marked as answer]
This should be fixed now, we execute command son tip with `<command>; exit` in your login shell. This is what iTerm and Terminal.app do as well.

**Reply by belizariogr** [reply]
Just tested and its fixed. Thank you very much!!
